app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb

```diff
@@ -129,10 +129,10 @@ def preload_question_test_cases
   end
 
   def extract_pathname_from_file(file_content, pathname)
-    class_name_extractor = /public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
+    class_name_extractor = /(^|;)\s*public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
     match = file_content.match(class_name_extractor)
 
-    match ? "#{match[1]}.java" : pathname.to_s
+    (!@question.multiple_file_submission && match) ? "#{match[1]}.java" : pathname.to_s
```

Comment:
This check on whether `question` is a `multiple_file_submission` should not the responsibility of `extract_pathname_from_file`